### PR TITLE
Include maven-release-plugin

### PR DESCRIPTION
**Pull request by pietrygamat**

### CODE REVIEW

**Comment by pietrygamat** [comment]
Introcudes release plugin. This helps keeping version in pom.xml in sync with git tagging (and by extension with github releases).

Executing `mvn release:prepare --batch-mode` will result in maven preparing the branch for release:
1. Update pom.xml version from SNAPSHOT to release
2. Commit the updated pom.xml
3. Tag the commit with matching label
4. Update pom.xml to next delevelopment version SNAPSHOT.

Because github workflow is configured to trigger on tag push, it will be performed as well.

File: .github/workflows/create-github-release.yml [hunk 1]
```diff
@@ -21,4 +21,4 @@ jobs:
         uses: softprops/action-gh-release@v1
         with: 
           files: target/Crucials-*.jar
-          prerelease: true
+          draft: true
```
Comment (pietrygamat) [inline .github/workflows/create-github-release.yml:24]: That was the original intention: gives some time for manual verification and approval before release is public.

File: pom.xml [hunk 2]
```diff
@@ -91,6 +94,16 @@
...
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-release-plugin</artifactId>
+                <version>3.0.0-M5</version>
+                <configuration>
+                    <allowTimestampedSnapshots>true</allowTimestampedSnapshots>
+                    <tagNameFormat>v@{project.version}</tagNameFormat>
+                    <pushChanges>false</pushChanges>
```
Comment (pietrygamat) [inline pom.xml:104]:
```
<pushChanges>false</pushChanges>
```
Executing `mvn release:prepare --batch-mode` creates commits and tag on current branch. The line above  can be removed, so the commit is also automatically pushed without user interaction.  
As long as `pushChanges` is set to false, the user has to manually perform `git push && git push --tags`, so there is time to fix any potential errors without triggering release in github.

**Comment by Tennessene** [comment]
Thanks for all the maven info and helping even more to make the process easier!

**Pull request merged**